src/components/sources/OlSourceWebglVector.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div>
+    <slot></slot>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VectorSource, { type Options } from "ol/source/Vector";
+import type { Ref } from "vue";
+import { inject, watch, onMounted, onUnmounted, provide, computed } from "vue";
+
+import Layer from "ol/layer/Layer.js";
+import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer.js";
+class WebGLVectorLayer extends Layer {
+  createRenderer() {
+    return new WebGLVectorLayerRenderer(this, {
+      className: this.getClassName(),
+      style: properties.style,
+    });
+  }
+}
+
+import type Line from "ol/geom/Line";
+import type Polygon from "ol/geom/Polygon";
+
+import usePropsAsObjectProperties from "@/composables/usePropsAsObjectProperties";
+import {
+  FEATURE_EVENTS,
+  useOpenLayersEvents,
+} from "@/composables/useOpenLayersEvents";
+
+// prevent warnings caused by event pass-through via useOpenLayersEvents composable
+defineOptions({
+  inheritAttrs: false,
+});
+
+const props = withDefaults(defineProps<Options<Line|Polygon>>(), {   // SHOULD prob. not be Line
```

Comment:
It should / could be [`SimpleGeometry`](https://openlayers.org/en/latest/apidoc/module-ol_geom_SimpleGeometry-SimpleGeometry.html) to be more flexible right?